### PR TITLE
Delete unused OverlayTree code from merge-tree package

### PR DESCRIPTION
**Pull request by curtisman**

### CODE REVIEW

File: packages/dds/merge-tree/src/test/tsconfig.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
         ],
     },
     "include": [
-        "./**/*"
+        "./**/*", "../text.ts"
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/test/tsconfig.json:13]: why is this needed?

**Review by anthony-murphy — APPROVED** [review]

File: packages/dds/merge-tree/src/test/tsconfig.json (hunk 1 above)
Comment (curtisman) [reply packages/dds/merge-tree/src/test/tsconfig.json:13]: ummm, I was moving things around and I think vscode added it... I will remove.
